pkg/asset/installconfig/aws/basedomain.go

```diff
@@ -95,5 +114,25 @@ func GetPublicZone(sess *session.Session, name string) (*route53.HostedZone, err
 	if res == nil {
 		return nil, errors.Errorf("No public route53 zone found matching name %q", name)
 	}
+
 	return res, nil
 }
+
+// GetPublicZoneByID returns a public route53 zone from the zone ID
+func GetPublicZoneByID(sess *session.Session, zoneID string) (*route53.HostedZone, error) {
+	// XXX: used for validation
+	if zoneID == "valid-public-zone-id" {
+		return &route53.HostedZone{
+			Id:   aws.String("valid-public-zone-id"),
+			Name: aws.String("valid-public-zone"),
+		}, nil
+	}
```

Comment:
I don't think we should include code in actual code-path that is only used for tests.. i think we should make the calling function testable by allowing us to replace that function with a fake that we can control output during tests.

Reply:
I have addressed this by converting the validation to use the go mock framework. I've done just enough to handle this case, however, I would like to convert the rest of the validation testing to use it as well in the future.